<a href="https://colab.research.google.com/github/spockthompson/Project-2-Machine-Learning/blob/main/Project_2_Part_1_(Core).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# First choice: dataset 1
 - https://www.kaggle.com/datasets/deepu1109/star-dataset
 - This data set uses key features to predict what type of star it is.

In [1]:
import pandas as pd
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [2]:
df1 = pd.read_csv('/content/sample_data/6 class csv.csv')
df1.head()

Temperature (K)  Luminosity(L/Lo)  Radius(R/Ro)  Absolute magnitude(Mv)  \
0             3068          0.002400        0.1700                   16.12   
1             3042          0.000500        0.1542                   16.60   
2             2600          0.000300        0.1020                   18.70   
3             2800          0.000200        0.1600                   16.65   
4             1939          0.000138        0.1030                   20.06   

   Star type Star color Spectral Class  
0          0        Red              M  
1          0        Red              M  
2          0        Red              M  
3          0        Red              M  
4          0        Red              M

In [3]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 240 entries, 0 to 239
Data columns (total 7 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Temperature (K)         240 non-null    int64  
 1   Luminosity(L/Lo)        240 non-null    float64
 2   Radius(R/Ro)            240 non-null    float64
 3   Absolute magnitude(Mv)  240 non-null    float64
 4   Star type               240 non-null    int64  
 5   Star color              240 non-null    object 
 6   Spectral Class          240 non-null    object 
dtypes: float64(3), int64(2), object(2)
memory usage: 13.2+ KB


3. What is the target?
 - Star Type
4. What does one row represent? (A person?  A business?  An event? A product?)
 - Lumosity represents the level of light reflection in space from that specific star from earth.
5. Is this a classification or regression problem?
 - Classification
6. How many features does the data have?
 - One
7. How many rows are in the dataset?
 - 240 Rows
8. What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?
 - I don't see much issue with this set, but we won't know for sure until we try.

#Second choice: dataset 2
 - https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset
 - This data set uses key features to predict the probability that a person with have a stroke.

In [4]:
df2 = pd.read_csv('/content/healthcare-dataset-stroke-data.csv')
df2.head()

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1

In [5]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5110 entries, 0 to 5109
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 5110 non-null   int64  
 1   gender             5110 non-null   object 
 2   age                5110 non-null   float64
 3   hypertension       5110 non-null   int64  
 4   heart_disease      5110 non-null   int64  
 5   ever_married       5110 non-null   object 
 6   work_type          5110 non-null   object 
 7   Residence_type     5110 non-null   object 
 8   avg_glucose_level  5110 non-null   float64
 9   bmi                4909 non-null   float64
 10  smoking_status     5110 non-null   object 
 11  stroke             5110 non-null   int64  
dtypes: float64(3), int64(4), object(5)
memory usage: 479.2+ KB


3. What is the target?
 - The Target would be the Stroke (Y/N)
4. What does one row represent? (A person?  A business?  An event? A product?)
 - Heart Disease represents a yes or a no in either a 1 or 0.
5. Is this a classification or regression problem?
 - Classification
6. How many features does the data have?
 - One feature "Heart Disease"
7. How many rows are in the dataset?
 - Looks to be 5110
8. What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?
 - Maybe cleaning up some of the data.